database/scripts/generate_report.rb

```diff
@@ -25,20 +25,23 @@
 options[:report_name] = "buildfarm-report_#{DateTime.now.strftime("%Y-%m-%d_%H-%M")}.json" if options[:report_name].nil?
 
 def generate_report(report_name, exclude_set)
+    report_regressions_all = BuildfarmToolsLib::test_regressions_all(filter_known: true)
+    report_regressions_consecutive = BuildfarmToolsLib::test_regressions_today(filter_known: true, only_consistent: true, group_issues: true, report_regressions: report_regressions_all)
+    report_regressions_flaky = BuildfarmToolsLib::flaky_test_regressions(filter_known: true, group_issues: true, report_regressions: report_regressions_all)
+    
```

Comment:
Why was this moved outside the report creation below?

Reply:
`test_regressions_all` method returns every test regression that happened in the last build. `test_regressions_today` and `flaky_test_regressions` run that method internally.

This is an optimization to not fetch all test regressions twice, but use only one run for both